polynote-kernel/src/main/scala/polynote/kernel/dependency/DependencyManager.scala

```diff
@@ -0,0 +1,89 @@
+package polynote.kernel.dependency
+
+import java.io.File
+import java.net.{URL, URLClassLoader}
+
+import polynote.config.RepositoryConfig
+import polynote.kernel.util.{LimitedSharingClassLoader, Publish}
+import polynote.kernel.{KernelStatusUpdate, TaskInfo}
+
+import scala.reflect.ClassTag
+import scala.reflect.internal.util.AbstractFileClassLoader
+import scala.reflect.io.AbstractFile
+
+/**
+  * A [[DependencyManager]] handles the fetching and setting up the provision of those dependencies (with a [[DependencyProvider]]).
+  *
+  * The idea is that when we want some dependencies to be available to our code, we typically need to do two things.
+  * First, we need to get those dependencies (and _their_ dependencies, too) from somewhere and (usually) download them
+  * as files onto our machine.
+  * Second, we need to somehow link those dependencies on our filesystem with our running code.
+  *
+  * An example of the former would be something like Coursier which download jars from remote repos onto our machine
+  * (in fact, we have a [[CoursierFetcher]]), while an example of the latter would be a classloader that provides the
+  * classes in said jars to our code (e.g., [[ClassLoaderDependencyProvider]]).
+  */
+trait DependencyManager[F[_]] {
```

Comment:
Needs a `FactoryBeanManagerFactory` 😛

Reply:
I went through a bunch of iterations for the names.. happy to hear any suggestions!